airflow/providers/amazon/aws/transfers/sql_to_s3.py

```diff
@@ -223,12 +223,9 @@ def _partition_dataframe(self, df: pd.DataFrame) -> Iterable[tuple[str, pd.DataF
         for group_label in (grouped_df := df.groupby(**self.groupby_kwargs)).groups:
             yield (
                 cast(str, group_label),
-                cast(
-                    "pd.DataFrame",
-                    grouped_df.get_group(group_label)
-                    .drop(random_column_name, axis=1, errors="ignore")
-                    .reset_index(drop=True),
-                ),
+                grouped_df.get_group(group_label)
```

Comment:
Related?

Reply:
Yes, someow it seems Python 3.9 improved typing, such `mypy-providers` complained with:
```
airflow/providers/amazon/aws/transfers/sql_to_s3.py:226: error: Redundant cast
to "DataFrame"  [redundant-cast]
                    cast(
                    ^
```